File: examples/python/basic-langchain-agent/src/agent.py
from tools import llm_tools
from llm import get_llm
from prompt import system_prompt
from frisk_sdk.adapters.langchain import FriskLangchain as Frisk
from langchain.agents import create_agent, AgentState
from langgraph.checkpoint.memory import InMemorySaver


class MyAgentState(AgentState):
    user_id: str
    redact_me: str


async def build_agent(
    frisk: Frisk,
):
    agent = create_agent(
        model=get_llm(),
        tools=await frisk.wrap_tools(llm_tools),
        system_prompt=system_prompt,
        state_schema=MyAgentState,
        middleware=[frisk.guard()],
        checkpointer=InMemorySaver(),
    )
    return agent

File: examples/typescript/basic-langchain-agent/src/agent.ts
import type { Frisk } from '@friskai/frisk-js/langchain'
import type { DynamicStructuredTool } from '@langchain/core/tools'
import {
  MemorySaver,
  StateSchema,
    MessagesValue
} from '@langchain/langgraph'
import { createAgent } from 'langchain'
import type { ZodObject } from 'zod'
import { z } from 'zod'
import { getLLM } from './llm.js'
import { systemPrompt } from './prompt.js'
import { llmTools } from './tools.js'

export const agentStateSchema = new StateSchema({
  messages: MessagesValue,
  userId: z.int(),
  redactMe: z.string().default(''),
})

export function buildAgent(frisk: Frisk) {
  const agent = createAgent({
    model: getLLM(),
    tools: frisk.wrapTools(
      llmTools as Iterable<DynamicStructuredTool<ZodObject>>,
    ),
    systemPrompt,
    stateSchema: agentStateSchema,
    middleware: [
      frisk.guard({ stateSchema: agentStateSchema })
    ],
    checkpointer: new MemorySaver(),
  })

  return agent
}
